main.go

```diff
@@ -376,25 +376,27 @@ func mainerr() error {
 				target += ".exe"
 			}
 
-			// optimistically remove our target in case we are installing over self
-			// TODO work out what to do for Windows
-			if mp.ImportPath == "github.com/myitcv/gobin" {
-				_ = os.Remove(target)
-			}
-
-			var stdout bytes.Buffer
-
-			installCmd := goCommand("install")
-			if *fTags != "" {
-				installCmd.Args = append(installCmd.Args, "-tags", *fTags)
-			}
-			installCmd.Args = append(installCmd.Args, mp.ImportPath)
-			installCmd.Dir = pkg.wd
-			installCmd.Env = append(buildEnv(localCacheProxy), "GOBIN="+gobin)
-			installCmd.Stdout = &stdout
-
-			if err := installCmd.run(); err != nil {
-				return err
+			// Only install if the target (within the gobin cache) does not
+			// exist. For now this cache is not read-only so this isn't as
+			// safe as it could/should be, but people shouldn't be messing with
+			// the cache anyway. The target in the cache is already hash by
+			// build tags so we should never have "overlapping" gobin runs.
+			if _, err := os.Stat(target); err != nil {
```

Comment:
why not `os.IsNotExist`?

you probably want to catch the "non-nil, not IsNotExist" case. For example, if it exists but it's not readable, we should error.

Reply:
Good point, thanks.